full-service/src/db/txo.rs

```diff
@@ -638,26 +662,25 @@ impl TxoModel for Txo {
                 }
                 TXO_RECEIVED => {
                     txo_details.received_to_account = Some(account_txo_status.clone());
+                    // Get the subaddress details if assigned
```

Comment:
Moved back under TXO_RECEIVED because the subaddress should only be added from the perspective of a Txo getting received.